### PR TITLE
Response Application - Add fields for director/reference info

### LINKED ISSUE #3232: Feature - Applicant - Collect email & name & job title for two references for each application
# Description

**Is your feature request related to a problem? Please describe.**
We want to send micro-reference requests via email to each applicant's director, and one other reference. Collecting this information directly on the application will allow us to automate the email generation.

**Describe the solution you'd like**
Add fields to the Basic Info page of the application for Name, Email, and Job Title for the applicant's Director and one other reference. These fields will need to be submitted, validated, and saved to the database like any other basic info fields.

**Describe alternatives you've considered**
We've considered re-purposing the existing Reference model (possible using Description to save Job Title). However, there's the possibility that users could add more references via their profile, which would confuse any automated email sending.

### PR DESCRIPTION
**Pull request by gggrant**
Resolves #3232.

- [ ] Translated

### CODE REVIEW

File: resources/views/applicant/strategic_response_application/step_01/questions.html.twig [hunk 1]
```diff
@@ -4,135 +4,135 @@
     Applicant: Application Post - Questions
 
 ============================================================================= #}
-<div
```
Comment (tristan-orourke) [inline resources/views/applicant/strategic_response_application/step_01/questions.html.twig]: I think I made a few changes to this file since you branched off feature/strategic-response-application. I'll try to reconcile them.

File: resources/lang/en/applicant/application_template.php [hunk 2]
```diff
@@ -101,9 +101,15 @@
...
-    'language_label' => 'Select One',
+    'director_name_label' => 'Director Name',
+    'director_title_label' => 'Director Title',
+    'director_email_label' => 'Director Email',
+    'reference_name_label' => 'Reference Name',
+    'reference_title_label' => 'Reference Title',
+    'reference_email_label' => 'Reference Email',
+    'required_info_heading' => 'Required Information',
+    'director_copy' => 'Director',
+    'reference_copy' => 'Reference',
```
Comment (tristan-orourke) [inline resources/lang/en/applicant/application_template.php]: Lets move this into the strategic_response section of this lang file, since its only used there so far.

File: app/Services/Validation/StrategicResponseApplicationValidator.php [hunk 3]
```diff
@@ -13,6 +13,12 @@ protected function basicInfoSimpleRules()
     {
         return [
             'preferred_language_id' => ['required', 'exists:preferred_languages,id'],
+            'director_name' => ['required', 'string'],
+            'director_title' => ['required', 'string'],
+            'director_email' => ['required', 'string'],
```
[the remark this answers is not in the document]
Comment (gggrant) [reply app/Services/Validation/StrategicResponseApplicationValidator.php]: Thanks. I knew that...

File: resources/lang/en/applicant/application_template.php [hunk 4]
```diff
@@ -101,9 +101,15 @@
...
-    'language_copy' => 'Which language would you prefer for this application process?',
-    'language_label' => 'Select One',
+    'director_name_label' => 'Director Name',
+    'director_title_label' => 'Director Title',
+    'director_email_label' => 'Director Email',
+    'reference_name_label' => 'Reference Name',
+    'reference_title_label' => 'Reference Title',
+    'reference_email_label' => 'Reference Email',
+    'required_info_heading' => 'Required Information',
+    'director_copy' => 'Director',
```
Comment (tristan-orourke) [inline resources/lang/en/applicant/application_template.php]: Lets adapt the full question text: `What is your current director’s name and email? (We’ll need approval that you can be loaned to another team for a bit to help out.)`
And for the reference copy: `What is the name and email of an additional reference who can confirm your skill set? (Preferably a current or recent Government of Canada supervisor.)`

File: resources/lang/en/applicant/application_template.php [hunk 5]
```diff
@@ -101,9 +101,15 @@
     'veterans_title' => 'Veterans Claim',
     'veterans_content' => 'Are you a veteran or a member of the Canadian Armed Forces?',
     'veterans_label' => 'Select One',
-    'language_title' => 'Language Requirement',
-    'language_copy' => 'Which language would you prefer for this application process?',
-    'language_label' => 'Select One',
```
[the remark this answers is not in the document]
Comment (gggrant) [reply resources/lang/en/applicant/application_template.php]: Dupes.

File: database/factories/ApplicationFactory.php [hunk 6]
```diff
@@ -25,7 +25,13 @@
...
         'applicant_id' => function () {
             return factory(Applicant::class)->create()->id;
-        }
+        },
+        'director_name' => $faker->name(),
+        'director_title' => $faker->jobTitle(),
+        'director_email' => $faker->email(),
+        'reference_name' => $faker->name(),
+        'reference_title' => $faker->jobTitle(),
+        'reference_email' => $faker->email()
```
[the remark this answers is not in the document]
Comment (gggrant) [reply database/factories/ApplicationFactory.php]: Done.

File: resources/views/applicant/strategic_response_application/step_01/questions.html.twig (hunk 1 above)
Comment (gggrant) [reply resources/views/applicant/strategic_response_application/step_01/questions.html.twig]: Hokay.

File: resources/lang/en/applicant/application_template.php (hunk 2 above)
Comment (gggrant) [reply resources/lang/en/applicant/application_template.php]: Done.

File: resources/lang/en/applicant/application_template.php (hunk 4 above)
Comment (gggrant) [reply resources/lang/en/applicant/application_template.php]: Adapted.

File: resources/views/applicant/strategic_response_application/step_01/questions.html.twig [hunk 7]
```diff
@@ -4,135 +4,132 @@
...
+                        </div>
+                    </div>
+                </div>
             </div>
-
         </div>
-
         <h3>
             {{ application_template.question_title }}
         </h3>
```
Comment (tristan-orourke) [inline resources/views/applicant/strategic_response_application/step_01/questions.html.twig]: I think we can actually remove this title ("My Fit") and just leave everything under "Required Information".

File: resources/lang/en/applicant/application_template.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (tristan-orourke) [reply resources/lang/en/applicant/application_template.php]: Ah, good point!

File: resources/views/applicant/strategic_response_application/step_01/questions.html.twig (hunk 7 above)
Comment (gggrant) [reply resources/views/applicant/strategic_response_application/step_01/questions.html.twig]: Done.

File: resources/assets/sass/components/applicant/application/post/_questions.scss [hunk 8]
```diff
@@ -33,11 +33,8 @@
 
     p {
         @extend %paragraph;
-        margin: calc(#{$unit} * .5) 0 calc(#{$unit} * 1);
-    }
-
-    button {
-
+        margin: calc(#{$unit} * 1) 0 calc(#{$unit} * 1);
+        padding-top: 1rem;
```
Comment (tristan-orourke) [inline resources/assets/sass/components/applicant/application/post/_questions.scss]: Why did you remove the button formatting?
